### ISSUE TITLE
`cargo insta test` exits with exit code `0` even if there's a test failure

### ISSUE THREAD

**Issue by MichaReiser** [mitsuhiko/insta#392]
### What happened?

I ran `cargo insta test` in a project with a test failure. `cargo insta` v1.29.0` returned an exit code 1, whereas the latest version returns an exit code of 0. This causes our CI to pass, even if there are test failures. 

### Reproduction steps

1. Create a project with a failing test, for example:

	```rust
	#[cfg(test)]
	mod tests {
	    #[test]
	    fn it_works() {
	        assert_eq!(3, 2);
	    }
	}
	```

2. Run `cargo insta test`
3. Print the exit code: `echo $status` -> 0

This seems to have changed with insta 1.30

1. Install insta 1.29.0: `cargo install cargo-insta@1.29.0`
2. Run `cargo insta test`
3. Print the exit code: `echo $status` -> 1

		

### Insta Version

1.30.0

### rustc Version

1.68.0

### What did you expect?

`cargo insta` to exit with code 1

**Comment by mitsuhiko** [comment]
This is now fixed on master. will be published soon.

**Comment by MichaReiser** [comment]
Thanks!

**Comment by not-my-profile** [comment]
I also ran into this bug today. Is there anything delaying the release of the fix? You may also want to consider yanking v1.30.0 since this is a pretty significant bug.

**Comment by mitsuhiko** [comment]
Sorry for the delay on this. It's out now.
